src/ODataProvider.cs

```diff
@@ -129,6 +130,27 @@ public string DestinationEndpointId
     #endregion
 
     private string GetMetadataURL()
```

Comment:
I think we can refactor this a bit, to something like

`    private string GetMetadataURL()
    {
        if (GetEndpointResponse(ODataSourceReader.GetEndpointUrlWithTop(_endpoint.Url), out string endpointResponse, out Exception exception) != HttpStatusCode.OK || exception is not null)
            return GetMetadataURLFallBack();

        using var responseJson = JsonDocument.Parse(endpointResponse);
        if (responseJson.RootElement.ValueKind != JsonValueKind.Object)
            return GetMetadataURLFallBack();

        return responseJson.RootElement.EnumerateObject().FirstOrDefault(obj => obj.Name.Equals("@odata.context", StringComparison.OrdinalIgnoreCase)).Value.GetString() ?? GetMetadataURLFallBack();
    }`

In general just avoiding the indentation, this also allows us to have more control of each fallback if we need to.

Reply:
fixed